syntax = "proto3";

package websocket;

message ClientToServerMessage {
  oneof payload {
    StartGameMessage start_game = 1;
  }
}

message StartGameMessage {
}

message ServerToClientMessage {
  oneof payload {
    LobbyUpdatedMessage lobby_updated = 1;
    GameStartedMessage game_started = 2;
    TimerUpdateMessage timer_update = 3;
  }
}

message LobbyUpdatedMessage {
  string room_code = 1;
  repeated string player_names = 2;
  bool is_host = 3;
}

message GameStartedMessage {
  string game_room_id = 1;
}

message TimerUpdateMessage {
  int32 remaining_seconds = 1;
}